docs/advanced_app_features.rst

```diff
@@ -119,3 +119,114 @@ which the child module forms do not. So we need to add it in:
+Shadow modules do not necessarily behave well when the source module uses custom case tiles. If you experience problems, make the shadow module's case tile configuration exactly matches the source module's.
+
+Entries
+~~~~~~~
+
+A shadow module duplicates all of its parent's entries. In the example below, m1 is a shadow of m0, which has one form. This results in two unique entries, one for each module, which share several properties.
+
+.. code-block:: xml
+
+    <entry>
+        <form>
+            http://openrosa.org/formdesigner/86A707AF-3A76-4B36-95AD-FF1EBFDD58D8
+        </form>
+        <command id="m0-f0">
+            <text>
+                <locale id="forms.m0f0"/>
+            </text>
+        </command>
+    </entry>
+    <entry>
+        <form>
+            http://openrosa.org/formdesigner/86A707AF-3A76-4B36-95AD-FF1EBFDD58D8
+        </form>
+        <command id="m1-f0">
+            <text>
+                <locale id="forms.m0f0"/>
+            </text>
+        </command>
+    </entry>
+
+Menu structure
+~~~~~~~~~~~~~~
+
+In the simplest case, shadow module menus look exactly like other module menus. In the example below, m1 is a shadow of m0. The two modules have their own, unique menu elements.
+
+.. code-block:: xml
+
+    <menu id="m0">
+        <text>
+            <locale id="modules.m0"/>
+        </text>
+        <command id="m0-f0"/>
+    </menu>
+    <menu id="m1">
+        <text>
+            <locale id="modules.m1"/>
+            </text>
+        <command id="m1-f0"/>
+    </menu>
+    
+
+Menus get more complex when shadow modules are mixed with parent/child modules. In the following example, m0 is a basic module, m1 is a child of m0, and m2 is a shadow of m0. All three modules have `put_in_root=false` (see **Child Modules > Menu structure** above).  The shadow module has its own menu and also a copy of the child module's menu. This copy of the child module's menu is given the id `m1.m2` to distinguish it from `m1`, the original child module menu.
+
+.. code-block:: xml
+
+    <menu id="m0">
+        <text>
+            <locale id="modules.m0"/>
+        </text>
+        <command id="m0-f0"/>
+    </menu>
+    <menu root="m0" id="m1">
+        <text>
+            <locale id="modules.m1"/>
+        </text>
+        <command id="m1-f0"/>
+    </menu>
+    <menu root="m2" id="m1.m2">                                                                                                     <text>
+            <locale id="modules.m1"/>
+        </text>                                                                                                                     <command id="m1-f0"/>
+    </menu>
+    <menu id="m2">                                                                                                                  <text>
+            <locale id="modules.m2"/>
+        </text>                                                                                                                     <command id="m2-f0"/>
```

Comment:
Yep, it does not duplicate the child module's entries, only the parent module's entries.